Templates/Per Tenant Extension/.github/workflows/UpdateGitHubGoSystemFiles.yaml

```diff
@@ -77,15 +77,25 @@ jobs:
           directCommit: '${{ github.event.inputs.directCommit }}'
           downloadLatest: ${{ github.event.inputs.downloadLatest }}
           eventName: ${{ github.event_name }}
+          GH_TOKEN: ${{ github.token }}
         run: |
-          $errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+          $errorActionPreference = "Stop";
+          $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
           $directCommit = $ENV:directCommit
           $downloadLatest = $ENV:downloadLatest
           Write-Host $ENV:eventName
           if ($ENV:eventName -eq 'schedule') {
-            Write-Host "Running Update AL-Go System Files on a schedule. Setting DirectCommit and DownloadLatest to true"
-            $directCommit = 'true'
+            Write-Host "Running Update AL-Go System Files on a schedule."
+            $rules = gh api /repos/$env:GITHUB_REPOSITORY/rules/branches/$env:GITHUB_REF | ConvertFrom-Json
```

Comment:
What about branch protection rules? https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28#get-branch-protection

These can fail the direct push as well.

Reply:
I tried a number of different branch protection rules and they all resulted in the pull request rule to be added to the rules - which is why I just test that instead of testing for an array of different things,

Reply:
Hm... weird. Can you double check? It makes sense for the API to handle both rulesets and branch protection rules, but I couldn't make it work. Why am I missing?

Here's what I did:
I added a branch protection rule for _main_:
![image](https://github.com/microsoft/AL-Go/assets/43066499/bdd0e288-710a-4e82-ba22-d783cf048e52)

Then 
```
gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/mazhelez/BCApps/rules/branches/main
``` 
returns _nothing_.

However, 
```
gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/mazhelez/BCApps/branches/main/protection
```
 returns an object with the following:

```
"required_pull_request_reviews": {
    "url": "https://api.github.com/repos/mazhelez/BCApps/branches/main/protection/required_pull_request_reviews",
    "dismiss_stale_reviews": false,
    "require_code_owner_reviews": false,
    "require_last_push_approval": false,
    "required_approving_review_count": 1
  }
```

Reply:
Now it will also check for branch protection and use PR unless it can bypass the protection rule.